services-hll/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/AsyncTable.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.enhanced.dynamodb.model.RequestItem;
+import software.amazon.awssdk.enhanced.dynamodb.model.ResponseItem;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.model.GetItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+
+/**
+ * An asynchronous way of interacting with a DynamoDB table. This is usually created via
+ * {@link DynamoDbEnhancedAsyncClient#table(String)}.
+ *
+ * This provides facilities for writing {@link RequestItem}s to a table and reading {@link ResponseItem}s from a table. All
+ * operations are non-blocking, returning a {@link CompletableFuture} for the result, instead of the result itself.
+ *
+ * Supported operations:
+ * <ol>
+ *     <li>{@link #getItem(RequestItem)} to retrieve a single item from the table.</li>
+ *     <li>{@link #putItem(RequestItem)} to write a single item to a table.</li>
+ * </ol>
+ *
+ * This {@link AsyncTable} will throw exceptions from all operations if the {@link DynamoDbEnhancedAsyncClient} that was used to
+ * create it is closed.
+ */
+@SdkPublicApi
+@ThreadSafe
+@Immutable
+public interface AsyncTable {
+    /**
+     * Retrieve the name of the DynamoDB table. This does not make a service call, and does not guarantee that the DynamoDB table
+     * actually exists.
+     */
+    default String name() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Call DynamoDB to write the requested item, potentially replacing an existing item that has a key matching this item. If
+     * this does not throw an exception, the item was successfully written.
+     *
```

Comment:
nit, missing the `<p>` tags. Seems all javadocs are missing it.

Reply:
Yeah, we're pretty inconsistent about this in all of our Javadoc. What happens without them? No separation?

Reply:
Yeah, there would be no separations without the p tag. 

https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#styleguide